### PR TITLE
cache keys

### LINKED ISSUE #122: Model objects should implement cache_key
Similar to [ActiveRecord's cache_key method](https://api.rubyonrails.org/classes/ActiveRecord/Integration.html#method-i-cache_key) we should implement cache_key and cache_version methods.  This allows them to be used as keys for [fragment caching](https://guides.rubyonrails.org/caching_with_rails.html#fragment-caching).  We ought to be able to cache an entire page's render tree based on the `Page` object's `cache_key`.

See https://github.com/rails/rails/blob/master/activesupport/lib/active_support/cache.rb for how these methods are used in Rails caching.

### PR DESCRIPTION
**Pull request by gburgett**
Add cache_key and cache_version to ActiveRecordShim

fixes #122

### CODE REVIEW

File: wcc-contentful/lib/wcc/contentful/active_record_shim.rb [hunk 1]
```diff
@@ -7,6 +7,26 @@ def attributes
     @attributes ||= to_h['fields'].tap { |fields| fields['id'] = id }
   end
 
+  def cache_key
+    key = "#{self.class.model_name}/#{id}"
+    key += "-#{cache_version}" unless ActiveRecord::Base.try(:cache_versioning) == true
+    key
+  end
+
+  def cache_key_with_version
```
Comment (jpowell) [inline wcc-contentful/lib/wcc/contentful/active_record_shim.rb]: why 2 methods?

Comment (gburgett) [reply wcc-contentful/lib/wcc/contentful/active_record_shim.rb]: I was following this blog: https://blog.bigbinary.com/2019/08/06/rails-adds-support-for-recyclable-cache-keys.html

Comment (gburgett) [reply wcc-contentful/lib/wcc/contentful/active_record_shim.rb]: https://api.rubyonrails.org/classes/ActiveRecord/Integration.html#method-i-cache_key_with_version

File: wcc-contentful/lib/wcc/contentful/active_record_shim.rb [hunk 2]
```diff
@@ -7,6 +7,26 @@ def attributes
     @attributes ||= to_h['fields'].tap { |fields| fields['id'] = id }
   end
 
+  def cache_key
+    key = "#{self.class.model_name}/#{id}"
+    key += "-#{cache_version}" unless ActiveRecord::Base.try(:cache_versioning) == true
```
Comment (jpowell) [inline wcc-contentful/lib/wcc/contentful/active_record_shim.rb]: why not just call cache key with version in this instance and return early

**Pull request merged**